### PR TITLE
Add ARM64 support

### PR DESCRIPTION
**Pull request by harshithmohan**

### CODE REVIEW

**Comment by Cazzar** [comment]
Is it better to add an arm specific docker file?

**Comment by harshithmohan** [comment]
> Is it better to add an arm specific docker file?

It works on x86 also, just debian got changed to ubuntu. It has to be the same dockerfile if we have to push both archs to the same tag (like in the latest commit).
Or I can do it separate it that's better

**Comment by harshithmohan** [comment]
image
Looks like this on docker hub (that's my repo)

**Comment by Cazzar** [comment]
> image Looks like this on docker hub (that's my repo)

TBH that's just the main part of tagging  - having the seperated docker files means that if we need to make differences to the builds, it is easier.

**Comment by Cazzar** [comment]
For naming conventions, can we look at changing to `Dockerfile.{arch}` as a standard naming - as this is used a lot in other places, ex: https://github.com/microsoft/docker

File: .github/workflows/docker-daily.yml [hunk 1]
```diff
@@ -6,17 +6,49 @@ on:
...
+    name: Build docker image for amd64
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@master
       with:
         submodules: recursive
     - name: Build the Docker image
-      run: docker build . --file Dockerfile --tag ${{ secrets.DOCKER_REPO }}:daily
+      run: docker build . --file Dockerfile --tag ${{ secrets.DOCKER_REPO }}:daily-amd64
```
Comment (Cazzar) [inline .github/workflows/docker-daily.yml:18]: can we push to the same tag, just the different arch.

File: .github/workflows/docker-daily.yml [hunk 2]
```diff
@@ -6,17 +6,49 @@ on:
...
     - name: Log into docker hub
       run: docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }}
     - name: Push daily to Docker Hub
-      run: docker push ${{ secrets.DOCKER_REPO }}:daily
+      run: docker push ${{ secrets.DOCKER_REPO }}:daily-amd64
```
Comment (Cazzar) [inline .github/workflows/docker-daily.yml:22]: can we push to the same tag, just the different arch.

**Comment by harshithmohan** [comment]
huh, I didn't know. I will change it

File: .github/workflows/docker-daily.yml [hunk 3]
```diff
@@ -6,17 +6,49 @@ on:
...
+    - uses: docker/setup-qemu-action@v1
+      name: Set up QEMU
+      with:
+        platforms: arm64
+    - uses: docker/setup-buildx-action@v1
+      name: Set up Docker Buildx
+    - name: Log into docker hub
+      run: docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }}
+    - name: Build and Push the Docker image
+      run: docker buildx build . --file Dockerfile_arm --tag ${{ secrets.DOCKER_REPO }}:daily-arm64 --platform linux/arm64 --push
```
Comment (Cazzar) [inline .github/workflows/docker-daily.yml:41]: can we push to the same tag, just the different arch.

File: .github/workflows/docker-manual.yml [hunk 4]
```diff
@@ -13,16 +13,48 @@ on:
...
         ref: "${{ github.event.inputs.ref }}"
     - name: Build the Docker image
-      run: docker build . --file Dockerfile --tag ${{ secrets.DOCKER_REPO }}:${{ github.event.inputs.tag }}
+      run: docker build . --file Dockerfile --tag ${{ secrets.DOCKER_REPO }}:${{ github.event.inputs.tag }}-amd64
     - name: Log into docker hub
       run: docker login --username ${{ secrets.DOCKER_USERNAME }} --password ${{ secrets.DOCKER_PASSWORD }}
-    - name: Push daily to Docker Hub
-      run: docker push ${{ secrets.DOCKER_REPO }}:${{ github.event.inputs.tag }}
+    - name: Push to Docker Hub
+      run: docker push ${{ secrets.DOCKER_REPO }}:${{ github.event.inputs.tag }}-amd64
```
Comment (Cazzar) [inline .github/workflows/docker-manual.yml:29]: can we push to the same tag, just the different arch.

**Review by Cazzar — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-manual.yml (hunk 4 above)
Comment (harshithmohan) [reply .github/workflows/docker-manual.yml:29]: We cannot, we have to push the tags separately and then combine them with manifest
We can delete them after manifest creation

Comment (harshithmohan) [reply .github/workflows/docker-manual.yml:29]: Let me try something

**Comment by harshithmohan** [comment]
@Cazzar Looks like there's no way. We have to push the separate tags and then link them with the manifest file to the `daily` tag. If we push to the same tag, it just overwrites the existing image.

**Comment by harshithmohan** [comment]
I could only think of some hacky ways to do it. Also, even the dotnet sdk images are in this way. Separate tags for each arch and then combined under a single tag for convenience.

**Comment by harshithmohan** [comment]
image
this is what it looks like currently, btw

**Comment by harshithmohan** [comment]
Well, this PR is messed up.. so I'll make another

**Pull request closed without merging**